clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/security/OpenstackCredentialsInitializer.groovy

```diff
@@ -0,0 +1,85 @@
+import com.netflix.spinnaker.clouddriver.security.ProviderUtils
+import org.apache.log4j.Logger
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Scope
+import org.springframework.stereotype.Component
+
+
+@Component
+@Configuration
+class OpenstackCredentialsInitializer implements CredentialsInitializerSynchronizable {
+  private static final Logger LOG = Logger.getLogger(this.class.simpleName)
+
+  @Autowired
+  AccountCredentialsRepository accountCredentialsRepository
+
+  @Autowired
+  ApplicationContext appContext
+
+  @Bean
+  List<? extends OpenstackNamedAccountCredentials> openstackNamedAccountCredentials(
+    OpenstackConfigurationProperties openstackConfigurationProperties) {
+    synchronizeOpenstackAccounts(openstackConfigurationProperties, null)
+  }
+
+  @Override
+  String getCredentialsSynchronizationBeanName() {
+    return "synchronizeOpenstackAccounts"
+  }
+
+  @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+  @Bean
+  List<?> synchronizeOpenstackAccounts(OpenstackConfigurationProperties openstackConfigurationProperties, CatsModule catsModule) {
+    def (ArrayList<OpenstackConfigurationProperties.ManagedAccount> accountsToAdd, List<String> namesOfDeletedAccounts) =
+      ProviderUtils.calculateAccountDeltas(accountCredentialsRepository,
+                                          OpenstackNamedAccountCredentials,
+                                          openstackConfigurationProperties.accounts)
+
+    accountsToAdd.each { OpenstackConfigurationProperties.ManagedAccount managedAccount ->
+      try {
+        def openstackAccount = new OpenstackNamedAccountCredentials(managedAccount.name,
+                                                                    managedAccount.environment ?: managedAccount.name,
+                                                                    managedAccount.accountType ?: managedAccount.name,
+                                                                    managedAccount.master,
+                                                                    managedAccount.endpoint,
+                                                                    managedAccount.tenantId,
+                                                                    managedAccount.username,
+                                                                    managedAccount.password)
+        accountCredentialsRepository.save(managedAccount.name, openstackAccount)
+      } catch (e) {
+        LOG.info "Could not load account ${managedAccount.name} for Openstack.", e
+      }
+    }
+    ProviderUtils.unscheduleAndDeregisterAgents(namesOfDeletedAccounts, catsModule)
+
```

Comment:
I would leave a `todo` here as a reminder that you need to add a call to `synchronizeAgentProviders` here later (see [this](https://github.com/spinnaker/clouddriver/pull/214/files#diff-3032472a8dba760597616f41ebbaf4e7R92)). The reason why it can't be added now is that you haven't registered any agents with openstack, and it will ultimately allow clouddriver config to be refreshed automatically without restarting clouddriver.